docs/modules/automation-testing/pages/scripting/create-image-injection-script.adoc

```diff
@@ -0,0 +1,98 @@
+== Appium capability (instrumentation only)
+
+Add the `'kobiton:instrumentenabled'` capability to your script’s capabilities list. The value of this capability must be `true` to instrument the app for image injection.
+
+
+[IMPORTANT]
+Only include this Appium capability for image injection via app instrumentation. *Do not* include it for apps with image injection SDK embedded.
+
+== Kobiton commands
+
+To automate the process of image injection, use commands provided by the Kobiton SDK.
+
+[options="header"]
+|=======================
+|Command | Description | Parameters
+| `kobiton:setImage` | Starts the image injection process by injecting a specified image into the camera view. | `data`: A base64 encoded string of the image you want to inject.
+| `kobiton:clearImage` |  This command stops the image injection process and reverts the camera view back to the live feed.
+ |
+|=======================
+
+=== Handle image file and injection
+
+Ensure the image file you intend to inject is stored locally and accessible by the test script.
+
+Pass the image file to the test script using `base64` encoding and include the command to inject it. Below is an example of an `injectImage` function demonstrating this process.
+
+.Example (javascript)
+[source,javascript]
+
+----
+
+async function injectImage(driver, imagepath) {
+  // assume the image is in the same folder as the test script so only need to provide a relative path to the image
+  const imagePath = path.join(__dirname, imagepath);
+
+  const base64Image = fs.readFileSync(imagePath, { encoding: 'base64' });
+
+  // the command to set image on the device
+  await driver.execute('kobiton:setImage', { data: base64Image });
+}
+
+----
+
+=== Start Image Injection
+
+Given that the `injectImage` function was already declared in the previous section, use it to begin the image injection process:
+
+.Example (javascript)
+[source,javascript]
+
+    await injectImage(driver, 'path_to_image/image.jpg');
+
+This will replace the camera view with the specified file in the relative path `path_to_image/image.jpg`.
+
+=== Stop image injection
+
+To stop image injection use:
+
+.Example (javascript)
+[source,javascript]
+
+    await driver.execute('kobiton:clearImage');
+
+This will revert the camera view back to the live feed.
+
+== Limitations
+
+* Only supported in Xium automation sessions.
+
+* The image must be located on the runner machine and be passed with `base64` encoding. Passing the image URL is not yet supported.
+
+include::apps:partials/image-injection-limitations.adoc[]
+
+== Known issues
+
+The Android application with custom SDK crashes when attempting to start the camera services.
+
+*Cause*: The custom SDK utilizes hidden APIs from Android, which may not be turned on and cause the app to crash.
+
+*Solution*: Enable a Cleanup Policy with *Device settings will be reset* option checked. If no Cleanup Policy is enabled, run the following `adb shell` command before launching the app:

```

Comment:
Can they do this command when in a script? I see this is in Manual session. Is there another solution when in automation?

Reply:
They can do it in an appium script. For the details on how to do that, they have to research themselves (since the audiences are android automation tester).

Reply:
This page is about automation. Are you saying the workaround is to open a Manual session to perform the adb shell command and then go run the script? If so - can you specify that the solution needs to be done in a Manual session? Thanks!

Reply:
No, there is guides out there to run adb shell commands right in the Appium automation script. Example: https://www.headspin.io/blog/running-arbitrary-adb-commands-via-appium. Do you want to give such a guide right in this page?

Reply:
Added the instructions anyway after verifying it myself. You can see the updated content at http://10.2.7.208/automation-testing/scripting/add-image-injection-to-appium-script

Reply:
GREAT! Thank you! Only change is to capitalize Javascript like this: JavaScript
Then publish!